src/components/menu/menu.types.ts

```diff
@@ -1,10 +1,50 @@
-import { MenuListItem } from '../menu-list/menu-list-item.types';
+import { ListSeparator } from '@limetech/lime-elements';
 
 export type OpenDirection = 'left' | 'right';
 
-export interface MenuItem<T = any> extends MenuListItem<T> {
+export interface MenuItem<T = any> {
     /**
      * The additional supporting text is used for shortcut commands and displayed in the list item.
      */
     commandText?: string;
+
+    /**
+     * Text to display in the list item.
+     */
+    text: string;
+
+    /**
+     * Additional supporting text to display in the list item.
+     */
+    secondaryText?: string;
+
+    /**
+     * True if the list item should be disabled.
+     */
+    disabled?: boolean;
+
+    /**
+     * Name of the icon to use.
+     */
+    icon?: string;
+
+    /**
+     * Background color of the icon. Overrides `--icon-background-color`.
+     */
+    iconColor?: string;
+
+    /**
+     * True if the list item should be selected.
+     */
+    selected?: boolean;
+
+    /**
+     * Value of the list item.
+     */
+    value?: T;
+
+    /**
+     * List of actions to display as a menu at the end of the item
+     */
+    actions?: Array<MenuItem | ListSeparator>;
```

Comment:
Can we have actions on a menu item?

Reply:
I guess it would make sense if we had sub-menus, but I don't think we do, so no, these should go. They'll be removed in a later commit though (possibly in a later PR even).

Reply:
Removing this made other things fail, so I'm moving that over to my step 3 branch. Whether we merge this first, or wait for that branch to merge into here, I'm good either way, I just want to keep this point as a fixed green build point that I won't accidentally fumble away 😂

Reply:
Okey! :P